tests/test_fixtures.py

```diff
@@ -32,19 +32,44 @@ def test_workspace_instance_scope_part2(workspace):
     assert workspace is workspace_instance[0]
 
 
-def test_hardware_node_add_remove(hardware):
+def _hardware_add_node(h, name, role) -> NodeBase:
     """
-    test the hardware fixture. Especially the handing of nodes
+    helper method to create a new hardware node
     """
-    nodes_length = len(hardware.nodes.keys())
+    nodes_length = len(h.nodes.keys())
     # create a new node
-    new_node = hardware.node_create('test1', NodeRole.WORKER, [])
+    new_node = h.node_create(name, role, [])
     # add the node to the hardware
-    hardware.node_add(new_node)
-    hardware.prepare_nodes(limit_to_nodes=[new_node])
+    h.node_add(new_node)
+    h.prepare_nodes(limit_to_nodes=[new_node])
     # we should have one more node now
-    assert len(hardware.nodes.keys()) == nodes_length+1
+    assert len(h.nodes.keys()) == nodes_length+1
+    return new_node
+
+
+def test_hardware_node_add_remove(hardware):
+    """
+    test the hardware fixture. Especially the handing of nodes
+    """
+    nodes_length = len(hardware.nodes.keys())
+    new_node = _hardware_add_node(hardware, 'test1', NodeRole.WORKER)
     # drop the node again
     hardware.node_remove(new_node)
     # we should have the old amount of nodes now
     assert len(hardware.nodes.keys()) == nodes_length
+
+
+def test_kubernetes_node_join(kubernetes):
+    """
+    test the kubernetes fixture
```

Comment:
Old comment?

Reply:
why?

Reply:
You're testing that a node can be added to kubernetes.

Reply:
I removed the comment